docs/DeveloperGuide.adoc

```diff
@@ -1,93 +1,1101 @@
+** Pros: Allows exams to be encapsulated using interface.
+** Cons: Temporary increase in space for object created during method .
+* **Alternative 2:** Have no interface.
+** Pros: Easy access to set values.
+** Cons: No encapsulation.
+
+===== Aspect: Data structure to support the exam commands
+
+* **Alternative 1 (current choice):** Exams are stored in a set under each Person object in the AddressBook but stored as a list in the ExamBook.
+** Pros: In the Exambook, it is required to access an exam through an index to delete and edit, hence list is better. In the AddressBook, exams are only to be used for viewing and accessed to check for value. 
+** Cons: Exams stored under each Person in the AddressBook cannot be sorted.
+* **Alternative 2:** Exams are stored as a list in both ExamBook and AddressBook.
+** Pros: Exams stored under each Person in the AddressBook can be sorted.
+** Cons: It will take longer to check if a Person has a same Exam with the Exam to be updated.
+
+===== Aspect: Storage of ExamBook
+
+* **Alternative 1 (current choice):** Exams are stored in a separate txt file from AddressBook
+** Pros: This allows for clarity and easy access to find and see exams in raw XML format. 
+** Cons: Extra storage file and hence methods and file paths are needed, this may be confusing and cause much repetition of code.
+* **Alternative 2:** Exams are stored in same data file as AddressBook.
+** Pros: Only 1 storage file is needed. Less repetition and variables are needed.
+** Cons: May be harder for user to see exams in raw XML format.
+// end::exam[]
+
+// tag::privilege[]
+=== Privilege Feature
+[[fig-PrivilegeDiagram]]
+.Structure of the Privilege Component
+image::Privilege.png[width="800"]
+
+==== Current Implementation
+Logic is assigned to one Privilege object at all times. Privilege contains a User and a Person.
+The User interface defines what access level the Privilege object have, which is implemented by BasicUser, TutorUser and AdminUser. +
+BasicUser is the class with the lowest access level, and the ancestor to other 2 User classes. +
+To create an increasing level of access, each User of a higher level inherits from the successively lower one. +
+Each User level have their own list of new commands they can run, which is appended to the list inherited from their parent. +
+Account class, which contains its respective Privilege object, is the way to assign a privilege level to a Person. +
+Account also holds authentication details required for logging in. + 
+When switching Privilege levels, the Privilege attached to Logic will point its UserType and myPerson to that in the desired Privilege level, as seen below in the Object Diagram and Sequence Diagram. +
+
+[[fig-PrivilegeObjectDiagram]]
+.How switching Privilege is implemented (Object Diagram)
+image::PrivilegeObjectDiagram.png[width="900"]
+
+[[fig-LoginSequenceDiagram]]
+.Sequence Diagram for Login Command
```

Comment:
Good job on the sequence diagram for the following:

- drew proper lifelines for each object
- specified invocation method
- use dashed arrows for return calls.